src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs

```diff
@@ -223,6 +223,26 @@ public async Task Handle(StorageInitializedNotification notification, Cancellati
             await EnsureInitializedAsync(cancellationToken);
         }
 
+        public IEnumerable<SearchParameterInfo> GetSearchParametersByResourceTypes(ICollection<string> resourceTypes)
+        {
+            return TypeLookup.Where(t => resourceTypes.Contains(t.Key)).SelectMany(t => t.Value.Values);
+        }
+
+        public IEnumerable<SearchParameterInfo> GetSearchParametersByUrls(ICollection<string> definitionUrls)
+        {
+            return UrlLookup.Where(t => definitionUrls.Contains(t.Key)).Select(t => t.Value);
+        }
+
+        public IEnumerable<SearchParameterInfo> GetSearchParametersByCodes(ICollection<string> codes)
+        {
+            return UrlLookup.Where(t => codes.Contains(t.Value.Code)).Select(t => t.Value);
+        }
+
+        public IEnumerable<SearchParameterInfo> GetSearchParametersByIds(ICollection<string> ids)
+        {
+            throw new NotImplementedException();
+        }
```

Comment:
Could these just be extension methods instead of changing the interface?

Reply:
Fair, since the other implementations just have these as not implemented exceptions. I can add this change to my PUT pr for this feature.